### ISSUE TITLE
Analytics Implementation or QA Support for Decision Review/AMA Part 2

### ISSUE THREAD

**Issue by MickinSahni** [department-of-veterans-affairs/va.gov-team#7345]
# Analytics Implementation or QA Request

## What this form is for
Use this template to request Google Tag Manager and Google Analytics implementation or QA for your product. Please fill out all brackets. 

> _Please see the following information for implementation timeline:_
> New Teams: Implementation will usually be completed in 4 sprints:
> 
> - Sprint 1: Discovery/Orientation
> - Sprint 2: FE Implementation
> - Sprint 3: Tagging Implementation
> - Sprint 4: QA
> 
> All other teams: Implementation will usually be completed in 2 sprints
> 
> - Sprint 1: Discovery and FE Implementation
> - Sprint 2: Tagging Implementation and QA

## Description
- Is this request for Google Analytics implementation or QA?
Google Analytics, potentially Foresee

- **Who is this request for?**
  - @MickinSahni  

- **What team/project is this work for?**
  -  Public Websites, Appeal Modernization Process (AMA)/Decision Part 2
    
- **Provide the Domain where Google Analytics needs to be implemented**
  - www.va.gov 
- **Please provide any upcoming deadlines or important milestones**
  - Pages are live in prod with CTAs. This request is to ensure those CTAs are being tracked along with Foresee 

- **Typical Website Interactions to Track**
   -  These pages were altered in some way and should have a) Foresee tracking enabled and b) analytics around the "Track the status of your claim" CTA:
       1. VA decisions reviews and appeals https://www.va.gov/decision-reviews/
       2. After you request a decision review https://www.va.gov/decision-reviews/after-you-request-review/
       3. File a Supplemental Claim with new evidence https://www.va.gov/decision-reviews/supplemental-claim/
       4. Request a Higher-Level Review https://www.va.gov/decision-reviews/higher-level-review/
       5. Request a Board Appeal https://www.va.gov/decision-reviews/board-appeal/
       6. After a Board Appeal ... https://www.va.gov/decision-reviews/board-appeal/after-board-appeal-decision/
       7. Multiple party contested claims https://www.va.gov/decision-reviews/multiple-party-claims/
       8. Fiduciary claims https://www.va.gov/decision-reviews/fiduciary-claims/
       9. Insurance claims https://www.va.gov/decision-reviews/insurance-claims/
       10. FAQs https://www.va.gov/decision-reviews/faq/
       11. Get help with... https://www.va.gov/decision-reviews/get-help-with-review-request/
       12. Board hearings with a VLJ https://www.va.gov/decision-reviews/board-appeal/veterans-law-judge-hearing/

   - Also, this page has certain links we should confirm are being tracked: https://www.va.gov/decision-reviews/
     - Supplemental Claim
       - https://www.vba.va.gov/pubs/forms/VBA-20-0995-ARE.pdf
       - https://www.va.gov/decision-reviews/supplemental-claim/
     - Higher Level Review  
        - https://www.vba.va.gov/pubs/forms/VBA-20-0996-ARE.pdf
        - https://www.va.gov/decision-reviews/higher-level-review/
      - Board Appeal
        - https://www.va.gov/vaforms/va/pdf/VA10182.pdf
        - https://www.va.gov/decision-reviews/board-appeal
- **Certain conditions/responses that need to be reviewed**

    - "Track the status of your claim" CTA
    - Foresee data

- **Additional Comments**
    - n/a
---
> Please leave the following blank

## Acceptance Criteria
- [ ] QA has been completed
- [ ] VFS team completes brief [VSP Collaboration Cycle Feedback](https://adhoc.optimalworkshop.com/questions/20260uu8-0-0/questions/before) survey

## Definition of Done
- [ ] All appropriate issue tagging is completed
- [ ] All AC completed

**Comment by jonwehausen** [comment]
#### Slack Chat with @MickinSahni 4.1
- Mickin's team is okay with leveraging page flow reporting for the additional links called out in the bottom half of the ticket
- Our scope will be to provide tracking on the "Track the status of your claim" button CTA

**Comment by MickinSahni** [comment]
> * Track the status of your claim

Thanks, @jonwehausen. Can you check my work?:
- [Flow report](https://analytics.google.com/analytics/web/?authuser=1#/report/content-engagement-flow/a50123418w177519031p176188361/_u.date00=20200101&_u.date01=20200331&_r.segment-0=www.va.gov~2Fdecision-reviews~2F&_r.segment-0-matchType=EXACT&_r.segment-0-name=www.v...iews~2F&_r.dimension=analytics.landingPagePath/) - Not sure I'm getting all outgoing links. I see /supplemental-claims/ though.
- [PDF Download](https://analytics.google.com/analytics/web/?authuser=1#/report/content-event-pages/a50123418w177519031p176188361/_u.date00=20200101&_u.date01=20200331&explorer-table.plotKeys=%5B%5D&_r.drilldown=analytics.pagePath:www.va.gov~2Fdecision-reviews~2F,analytics.eventCategory:Interactions/)

**Comment by jonwehausen** [comment]
### Datalayer Requirements

### Read Me - Developer Notes
- Please replicate the event on each of the pages listed where the CTA is housed

Interaction | Screenshot | Datalayer Requirements
------------ | -------------  | -------------
Clicks on "Track the status of your claim" CTA  | ![image](https://user-images.githubusercontent.com/48527022/78282293-0697c380-74ea-11ea-9305-c339afc40fd0.png) | ```dataLayer.push({```<br>```'event':'cta-primary-button-click'```<br>```});```  |

**Comment by jonwehausen** [comment]
@MickinSahni 

You are very close! Appreciate you giving that a try. For your convenience, I just created you a [custom report](https://analytics.google.com/analytics/web/template?uid=JYSzyOu-Tzy0ZMjphIEtSw) that you can go to that is filtered down for page hops to from https://www.va.gov/decision-reviews/:
- https://www.va.gov/decision-reviews/supplemental-claim/
- https://www.va.gov/decision-reviews/higher-level-review/
- https://www.va.gov/decision-reviews/board-appeal

When clicking on the [Custom Report template link](https://analytics.google.com/analytics/web/template?uid=JYSzyOu-Tzy0ZMjphIEtSw), be sure to find the "**VA.gov - Production**" Property and click on the **WBC - VA.gov Modernized Content** view. 

![image](https://user-images.githubusercontent.com/48527022/78286642-f504ea80-74ee-11ea-89d6-4247cc13ef4d.png)


Regarding the PDF downloads, please [use this report ](https://analytics.google.com/analytics/web/?authuser=1#/report/content-event-pages/a50123418w177519031p184624291/_u.date00=20200101&_u.date01=20200331&explorer-table.plotKeys=%5B%5D&explorer-table.advFilter=%5B%5B0,%22analytics.eventAction%22,%22RE%22,%22%5EPDF%20Downloaded%22,0%5D%5D&_r.drilldown=analytics.pagePath:~2Fdecision-reviews~2Findex.html&explorer-segmentExplorer.segmentId=analytics.eventAction/) as it's within the WBC modernized view and is filtered specifically for PDF Downloads. Moving forward, if this report gets funky, hopefully this screenshot and steps will help in quickly accessing again. 
1) Access the WBC - VA.gov Modernized view
2) Go to Events > Pages on the left navigation
3) Find your page = /decision-reviews/index.html and click on it
4) The breadcrumb at the top should change to PAGE = /decision-reviews/index.html
5) Toggle the Primary dimension to Event Action
6) Click the tiny edit button over to the right
7) Within the filter panels, set Include Event Action Matching RegExp = ^PDF Downloaded
8) Click Save 😄 

![image](https://user-images.githubusercontent.com/48527022/78285010-3ea10580-74ee-11ea-97de-ff7474b41adb.png)

**Comment by jonwehausen** [comment]
@joanneesteban Would you mind breaking out the Foresee portion portion of this request into it's own issue? Thanks in advance.

**Comment by MickinSahni** [comment]
Thanks so much, team! @jonwehausen wanted to quickly confirm on this one, our team would need to make the requisite changes on those pages, right? 

> ### Datalayer Requirements
> @MickinSahni Please find here the dataLayer requirements to support the "Track your claim status or appeal" clicks
> 
> ### Read Me - Developer Notes
> * Please replicate the event on each of the pages listed where the CTA is housed
> 
> Interaction	Screenshot	Datalayer Requirements
> Clicks on "Track the status of your claim" CTA	![image](https://user-images.githubusercontent.com/48527022/78282293-0697c380-74ea-11ea-9305-c339afc40fd0.png)	`dataLayer.push({`
> `'event':'cta-primary-button-click'`
> `});`

**Comment by jonwehausen** [comment]
Hi @MickinSahni,

Yes sir - we will need that dataLayer event be pushed on each of the pages you identified where that button lives.

**Comment by jonwehausen** [comment]
Hi @MickinSahni, 

If I may, we're going to make a slight adjustment to the event naming convention. Instead of ```cta-primary-button-click``` we would like to use ```cta-default-button-click```.

I have updated my comment [here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/7345#issuecomment-608025197) for your reference.

**Comment by joanneesteban** [comment]
## ForeSee Update

@MickinSahni - the comment card for ForeSee is updated on both staging & prod for each of these URLs, but please check prod after launch to ensure the survey still appears.

**Comment by MickinSahni** [comment]
> ## ForeSee Update
> @MickinSahni - the comment card for ForeSee is updated on both staging & prod for each of these URLs, but please check prod after launch to ensure the survey still appears.

Confirmed on all of the urls above (except for PDFs, of course :) )

**Comment by joanneesteban** [comment]
Blocking this via #7558

**Comment by jonwehausen** [comment]
Hi @MickinSahni, 

Could we get an update on the dataLayer push for this guy? Are we able to get an FE resource on https://github.com/department-of-veterans-affairs/va.gov-team/issues/7558? 

Thanks for your assistance

**Comment by joanneesteban** [comment]
Hey @stephorkandatx !

Wanted to bring this analytics tagging issue to your attention. Would we be able to get an FE resource for #7558 ?

Thanks!

**Comment by stephorkandatx** [comment]
@jonwehausen we will look at #7558 today in sprint planning. if we have questions or it remains blocked, we'll follow up. thanks!

**Comment by ncksllvn** [comment]
Hey @jonwehausen , I am implementing this by creating a generic "button link" listener. Basically, if there is a plain link that looks like a button, then this event is attached. It also adds a `buttonText` property to help us see what button links are getting clicked. For example -

```js
window.dataLayer.push({
   "event": "cta-default-button-click",
   "buttonText": "Track the status of your claim or appeal",
   "gtm.uniqueEventId": 23
})
```

Does this event look okay>

**Comment by ncksllvn** [comment]
Work here has been merged but I feature flagged it to not to go prod right now. It should go to staging in ~30 minutes though.

**Comment by jonwehausen** [comment]
Hi @ncksllvn , 

Apologies for the delayed response! Did a quick check of these in staging and looks great! 

We will have QA documented here shortly.

**Comment by jonwehausen** [comment]
### QA Testing
cc: @bsmartin-ep @ncksllvn @stephorkandatx 

Description of Interaction | dataLayer Screenshot | Tag Screenshot | Status
------------ | ------------- | ------------- | -------------
Click on button CTA - _Track the status of your claim_ - /fiduciary-claims page | ![image](https://user-images.githubusercontent.com/48527022/83407655-2d398400-a3df-11ea-9694-375d5038f4a1.png) | ![image](https://user-images.githubusercontent.com/48527022/83407859-93260b80-a3df-11ea-8288-0a84631b5a8c.png) | ✔️ **PASS**
Click on button CTA - _Track the status of your claim_ - /insurance-claims page | ![image](https://user-images.githubusercontent.com/48527022/83407992-dc765b00-a3df-11ea-8245-703310a25cc6.png) | ![image](https://user-images.githubusercontent.com/48527022/83408018-e730f000-a3df-11ea-9ec4-39aed84f348e.png) | ✔️ **PASS**
Click on button CTA - _Track the status of your claim_ - /supplemental-claim page | ![image](https://user-images.githubusercontent.com/48527022/83408280-61617480-a3e0-11ea-909b-8c2e89d31538.png) | ![image](https://user-images.githubusercontent.com/48527022/83408323-70482700-a3e0-11ea-8956-7f0c8c6641c3.png) | ✔️ **PASS**

**Comment by ncksllvn** [comment]
Oof yeah, this is why I enabled only staging for now!

The Pittsburgh one is confusing to me. The buttons are blue (which is the default button in the design system) but the event is `cta-primary-button-click`.

I'll start trying to think of a way to tweak this

**Comment by jonwehausen** [comment]
Yes, thanks @ncksllvn! Just want to make sure we are aligning where we can with the design system. Let me know if you need me to clear anything up.

**Comment by ncksllvn** [comment]
I see the issue with the Pittsburgh buttons triggering a double event, but it's confusing to me because those buttons are the default button type and they trigger the `cta-primary-button-click` event. Shouldn't they trigger the `cta-default-button-click `?

**Comment by jonwehausen** [comment]
Hi @ncksllvn , 

Yes, the Pittsburgh buttons should be triggering the `default` button click. I believe there may have been an issue on our part where we requested the wrong button was used. But yes, the blue ones should be `default` and the green should be `primary` as they are in the design system. 

Thank you Nick!

**Comment by ncksllvn** [comment]
Thanks Jon! Should I correct the Pittsburgh ones then while I'm at it?

**Comment by jonwehausen** [comment]
Yes please! That would be excellent. Would really appreciate that.

**Comment by ncksllvn** [comment]
Gotcha, I have a PR open - https://github.com/department-of-veterans-affairs/vets-website/pull/12883 - to add analytics to default button links, excluding those that are inside React widgets

**Comment by jonwehausen** [comment]
Thank you Nick, looks great! Our tagging has been published -- we are ready for this to move to production at the PRs approval.

**Comment by jonwehausen** [comment]
Hey @ncksllvn ,

Was going some validation and wasn't yet seeing the `cta-default-button-click` on the pages int the description in in production. I did see however the `cta-default-button-click` event corrected for Pittsburgh VAMC, thank you again for doing so. 

Should we expect to see the `cta-default-button-click` event in production on Monday with 3pm deploy?

Thanks,
Jon Wehausen

**Comment by ncksllvn** [comment]
Hey @jonwehausen, so I held off on merging my PR last week because the deployments were somewhat unpredictable and I wanted you to have time to test in Staging. I just merged it though - can you verify when you get the chance?

**Comment by jonwehausen** [comment]
Hey @ncksllvn ,

Thanks for the notice there. I just tested in [staging here](https://staging.va.gov/pittsburgh-health-care/) and was seeing the two button click events...

![image](https://user-images.githubusercontent.com/48527022/84054761-dd7c2f00-a981-11ea-9e45-ff214c33cfbf.png)


Should we be seeing just the `cta-default-button-click` event now in staging? Thanks a lot for all of your help here.

**Comment by ncksllvn** [comment]
@jonwehausen your find is totally valid - I updated one PIttsburgh template but missed a second one. Just opened a PR to update that second one as well, https://github.com/department-of-veterans-affairs/vets-website/pull/12997

**Comment by jonwehausen** [comment]
Great @ncksllvn thank you!
